packages/cli/src/commands/app.ts

```diff
@@ -1,210 +1,108 @@
-      );
-    } catch (e) {
-      print.error(e.message);
-      process.exitCode = 1;
-      return;
-    }
-
-    if (help) {
-      print.info(HELP);
-      return;
-    }
-
-    // Validate Params
-    const paramsValid = validateAppParams(
-      print,
-      command,
-      codegenDir,
-      ipfs,
-      ens
-    );
-
-    if (!paramsValid) {
-      print.info(HELP);
-      process.exitCode = 1;
-      return;
-    }
 
-    // Resolve manifest
-    const manifestPaths = manifestFile ? [manifestFile] : defaultAppManifest;
-    manifestFile = resolvePathIfExists(filesystem, manifestPaths);
+type AppCommandOptions = {
+  manifestFile: string;
+  codegenDir: string;
+  ipfs?: string;
+  ens?: string;
+};
 
-    if (!manifestFile) {
-      print.error(
-        intlMsg.commands_app_error_manifestNotFound({
-          paths: manifestPaths.join(", "),
+export const app: Command = {
+  setup: (program: Program) => {
+    const appCommand = program
+      .command("app")
+      .alias("a")
+      .description(intlMsg.commands_app_description());
+
+    appCommand
+      .command("codegen")
+      .description(intlMsg.commands_app_codegen())
+      .option(
+        `-m, --manifest-file <${intlMsg.commands_codegen_options_o_path()}>`,
+        intlMsg.commands_app_options_codegen({
+          default: defaultOutputTypesDir,
         })
-      );
-      return;
-    }
-
-    // Get providers and client
-    const { ipfsProvider, ethProvider } = await getTestEnvProviders(ipfs);
-    const ensAddress: string | undefined = ens;
-    const client: Web3ApiClient = getSimpleClient({
-      ensAddress,
-      ethProvider,
-      ipfsProvider,
-    });
-
-    // App project
-    const project = new AppProject({
-      rootCacheDir: path.dirname(manifestFile),
-      appManifestPath: manifestFile,
-      client,
-    });
-    await project.validate();
-
-    if (codegenDir) {
-      codegenDir = filesystem.resolve(codegenDir);
-    } else {
-      codegenDir = filesystem.resolve(defaultOutputTypesDir);
-    }
-
-    const schemaComposer = new SchemaComposer({
-      project,
-      client,
-    });
-    const codeGenerator = new CodeGenerator({
-      project,
-      schemaComposer,
-      outputDir: codegenDir,
-    });
-
-    if (await codeGenerator.generate()) {
-      print.success(`🔥 ${intlMsg.commands_app_success()} 🔥`);
-      process.exitCode = 0;
-    } else {
-      process.exitCode = 1;
-    }
+      )
+      .option(
+        `-c, --codegen-dir <${intlMsg.commands_codegen_options_o_path()}>`,
+        `${intlMsg.commands_app_options_codegen({
+          default: defaultOutputTypesDir,
+        })}`
+      )
+      .option(
+        `-i, --ipfs [<${intlMsg.commands_codegen_options_i_node()}>] `,
+        `${intlMsg.commands_codegen_options_i()}`
+      )
+      .option(
+        `-e, --ens [<${intlMsg.commands_codegen_options_e_address()}>]`,
+        `${intlMsg.commands_codegen_options_e()}`
+      )
+      .action(async (options) => {
```

Comment:
This manual parsing isn't needed but it seems like there is some issue with `commander` parser handler so until it gets fixed, we will be using manual parsing.